### ISSUE TITLE
n_jobs argument is ignored by Parallel

### ISSUE THREAD

**Issue by harishmk** [joblib/joblib#405]
Using joblib 0.10.2, gets this warning:

> E:\3rdParty\Enthought7.6.1\lib\site-packages\joblib\parallel.py:540: UserWarning: Multiprocessing-backed parallel loops cannot be nested, setting n_jobs=1
>   **self._backend_args)

```
import multiprocessing as mp
from math import sqrt

from joblib import Parallel, delayed

def calc(_):
    print Parallel(n_jobs=1)(delayed(sqrt)(i**2) for i in xrange(1))

#------------------------------------------------------------------------------
def main():
    pool = mp.Pool(processes=1)
    pool.imap_unordered(calc, xrange(1))
    pool.close()
    pool.join()
    return
#------------------------------------------------------------------------------
if __name__=='__main__':main()

```

**Comment by GaelVaroquaux** [comment]
To make this clear: the issue here would be the warning for n_job=1. Inner parallel computing would not work for n_jobs!=1, as multiprocessing cannot do nested.

However, I cannot reproduce the warning with the code that you have given here.

**Comment by harishmk** [comment]
Comes from here:
```
__parallel_backends.py: 262
    def effective_n_jobs(self, n_jobs):
        """Determine the number of jobs which are going to run in parallel.

        This also checks if we are attempting to create a nested parallel
        loop.
        """
        if mp.current_process().daemon:              # <----- this does not check if n_jobs == 1
            # Daemonic processes cannot have children
            warnings.warn(
                'Multiprocessing-backed parallel loops cannot be nested,'
                ' setting n_jobs=1',
                stacklevel=3)
            return 1
...
```

**Comment by lesteve** [comment]
Sorry @harishmk I am not sure what the problem is here. Are you saying that the warning is unclear? joblib.Parallel in a pool subprocess will switch to sequential execution (as you pointed out we are using mp.current_process().daemon to prevent nested parallelism).

**Comment by lesteve** [comment]
I missed the associated PR earlier, this makes more sense now, sorry for the noise!

**Comment by lesteve** [comment]
Closed by #406.

**Comment by lesteve** [comment]
@harishmk for next time, please remember to have "Fix #issueNumber" in your PR description. This way the issue gets closed automatically when the associated PR is merged.
